lib/transactional_system.pm

```diff
@@ -0,0 +1,119 @@
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: General library for every system that uses transactional-updates
+# Like CaasP, Kubic and transactional-server
+# Maintainer: Sergio Lindo Mansilla <slindomansilla@suse.com>
+
+package transactional_system;
+
+use base Exporter;
+use Exporter;
+
+use strict;
+use warnings;
+use testapi;
+use version_utils 'is_caasp';
+use power_action_utils 'power_action';
+
+our @EXPORT = qw(
+  rpmver
+  trup_call
+  trup_install
+);
+
+# Return names and version of packages for transactional-update tests
+sub rpmver {
+    my $q    = shift;
+    my $d    = get_var 'DISTRI';
+    my $arch = get_var('ARCH');
+
+    # package name | initial version
+    my %rpm = (
+        kubic => {
+            fn => '5-2.1',
+            in => '2.1',
+        },
+        caasp => {
+            fn => '5-5.3.61',
+            in => '5.3.61',
+        });
+
+    if ("$arch" eq "aarch64") {
+        %rpm = (
+            kubic => {
+                fn => '5-4.3',
+                in => '4.3',
+            });
+    }
+
+    # Returns expected package version after installation / update
+    if ($q eq 'in') {
+        return $rpm{$d}{$q};
+    }
+    # Returns rpm path for initial installation
+    else {
+        return " update-test-trival/update-test-$q-$rpm{$d}{fn}.$arch.rpm";
+    }
+}
+
+# Optionally skip exit status check in case immediate reboot is expected
+sub trup_call {
+    my $cmd   = shift;
+    my $check = shift // 1;
+    $cmd .= " > /dev/$serialdev";
+    $cmd .= " ; echo trup-\$?- > /dev/$serialdev" if $check;
+
+    script_run "transactional-update $cmd", 0;
+    if ($cmd =~ /pkg |ptf /) {
+        if (wait_serial "Continue?") {
+            send_key "ret";
+            # Abort update of broken package
+            if ($cmd =~ /\bup(date)?\b/ && $check == 2) {
+                die 'Abort dialog not shown' unless wait_serial('Abort');
+                send_key 'ret';
+            }
+        }
+        else {
+            die "Confirmation dialog not shown";
+        }
+    }
+    # Check if trup passed
+    wait_serial 'trup-0-' if $check == 1;
+    # Broken package update fails
+    wait_serial 'trup-1-' if $check == 2;
+}
+
+# Install a pkg in Kubic
+sub trup_install {
+    my $input = shift;
+    my ($unnecessary, $necessary);
+
+    # rebootmgr has to be turned off as prerequisity for this to work
+    script_run "rebootmgrctl set-strategy off";
+
+    my @pkg = split(' ', $input);
+    foreach (@pkg) {
+        if (!script_run("rpm -q $_")) {
+            $unnecessary .= "$_ ";
+        }
+        else {
+            $necessary .= "$_ ";
+        }
+    }
+    record_info "Skip", "Pre-installed (no action): $unnecessary" if $unnecessary;
+    if ($necessary) {
+        record_info "Install", "Installing: $necessary";
+        trup_call("pkg install $necessary");
+        process_reboot(1);
```

Comment:
It is funny, because it failed when I it was `process_reboot 1;`
But, after adding parentheses, it works properly -> http://slindomansilla-vm.qa.suse.de/tests/1090#step/transactional_update/139

Reply:
trup_install is used only by umoci,rootless,runc,skopeo modules which are not scheduled - I guess that's why there was no problem. But I am not sure if those modules are still used somewhere..
Maybe @drpaneas knows..